stream/src/public_key/handshake.rs

```diff
@@ -12,15 +12,15 @@ use std::time::Duration;
 /// Handshake information that is signed over by the sender.
 pub struct Info<C: Scheme> {
     /// The public key of the recipient.
-    recipient: C::PublicKey,
+    pub recipient: C::PublicKey,
 
     /// The ephemeral public key of the sender.
     ///
     /// This is used to derive the shared secret for the encrypted connection.
-    ephemeral_public_key: x25519::PublicKey,
+    pub ephemeral_public_key: x25519::PublicKey,
```

Comment:
I suppose an alternative here would be to just accept the public key in the constructor?

Reply:
We never use the private key, so I think this might be clearer.